lib/krane/kubernetes_resource/daemon_set.rb

```diff
@@ -87,5 +129,28 @@ def parent_of_pod?(pod_data)
       pod_data["metadata"]["ownerReferences"].any? { |ref| ref["uid"] == @instance_data["metadata"]["uid"] } &&
       pod_data["metadata"]["labels"]["pod-template-generation"].to_i == template_generation.to_i
     end
+
+    def required_rollout
+      krane_annotation_value("required-rollout") || DEFAULT_REQUIRED_ROLLOUT
+    end
+
+    def percent?(value)
+      value =~ /\d+%/
```

Comment:
I'd suggest more validation that the percent is valid than this regex (which is "one or more digits followed by a % character" as I understand it), eg, values over 100 are currently valid here.

Edit: looks like this is copied directly from https://github.com/Shopify/krane/blob/6e0a0097716a01d168652769178a90968d2a188d/lib/krane/kubernetes_resource/deployment.rb#L210

Maybe this and the other shared logic should be moved into a shared module that can be included on both classes? This could prevent the behavior unexpectedly diverging in the future